src/lambda/wizards/samDeployWizard.ts

```diff
@@ -299,35 +299,83 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
     /**
      * Retrieves an S3 Bucket to deploy to from the user.
      *
+     * @param selectedRegion Selected region for S3 client usage
      * @param initialValue Optional, Initial value to prompt with
      *
      * @returns S3 Bucket name. Undefined represents cancel.
      */
     public async promptUserForS3Bucket(selectedRegion: string, initialValue?: string): Promise<string | undefined> {
-        const inputBox = input.createInputBox({
+        const loadingBuckets: string = localize('AWS.samcli.deploy.s3bucket.picker.loading', 'Loading S3 buckets...')
+        const noBuckets: string = localize('AWS.samcli.deploy.s3bucket.picker.noBuckets', 'No buckets found.')
+        const bucketError: string = localize(
+            'AWS.samcli.deploy.s3bucket.picker.error',
+            'There was an error loading S3 buckets.'
+        )
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
             buttons: [this.helpButton, vscode.QuickInputButtons.Back],
             options: {
                 title: localize(
                     'AWS.samcli.deploy.s3Bucket.prompt',
                     'Enter the AWS S3 bucket to which your code should be deployed'
                 ),
+                value: initialValue,
+                matchOnDetail: true,
                 ignoreFocusOut: true,
-                prompt: localize(
-                    'AWS.samcli.deploy.s3Bucket.region',
-                    'S3 bucket must be in selected region: {0}',
-                    selectedRegion
-                ),
             },
+            items: [
+                {
+                    label: loadingBuckets,
+                },
+            ],
         })
 
-        // Pre-populate the value if it was already set
-        if (initialValue) {
-            inputBox.value = initialValue
-        }
+        quickPick.busy = true
+        quickPick.enabled = false
 
-        return await input.promptUser({
-            inputBox: inputBox,
-            onValidateInput: validateBucketName,
+        new Promise(async resolve => {
+            const goBack: string = localize('AWS.picker.dynamic.noItemsFound.detail', 'Click here to go back')
+
+            try {
+                const s3Client = new DefaultS3Client(
```

Comment:
Concrete Client implementations should not be directly instantiated. I believe all of the extension code goes through `ToolkitClientBuilder`

Reply:
I thought that was the case but for some reason we didn't have an S3 client as a part of that? Might be tied to the partition stuff, I'll check.